### PR TITLE
Balder/remove tld from config model rebased

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR
This is a rebase of #10742 with tests added.

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/search/IndexedSearchCluster.java [hunk 1]
```diff
@@ -355,6 +349,9 @@ public void getConfig(DispatchConfig.Builder builder) {
             nodeBuilder.fs4port(node.getDispatchPort());
             builder.node(nodeBuilder);
         }
+        if (useAdaptiveDispatch)
```
Comment (geirst) [inline config-model/src/main/java/com/yahoo/vespa/model/search/IndexedSearchCluster.java]: Please consider "{" on one-line if statements.

File: config-model/src/main/java/com/yahoo/vespa/model/search/IndexedSearchCluster.java [hunk 2]
```diff
@@ -369,6 +366,9 @@ public void getConfig(DispatchConfig.Builder builder) {
                     break;
             }
         }
+        if (tuning.dispatch.maxHitsPerPartition != null)
```
Comment (geirst) [inline config-model/src/main/java/com/yahoo/vespa/model/search/IndexedSearchCluster.java]: Please consider "{" on one-line if statements.

**Review by geirst — APPROVED** [review]
👍

File: config-model/src/main/java/com/yahoo/vespa/model/search/IndexedSearchCluster.java (hunk 1 above)
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/vespa/model/search/IndexedSearchCluster.java]: Normally I do, but I prefer following the consensus in the file. And here all where non-braced.

File: config-model/src/main/java/com/yahoo/vespa/model/search/IndexedSearchCluster.java (hunk 2 above)
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/vespa/model/search/IndexedSearchCluster.java]: Normally I do, but I prefer following the consensus in the file. And here all where non-braced.

**Pull request merged**